library/L1_Peripheral/stm32f10x/pwm.hpp

```diff
@@ -0,0 +1,446 @@
+#pragma once
+
+#include "L0_Platform/stm32f10x/stm32f10x.h"
+#include "L1_Peripheral/pwm.hpp"
+#include "L1_Peripheral/stm32f10x/pin.hpp"
+#include "L1_Peripheral/stm32f10x/system_controller.hpp"
+#include "utility/log.hpp"
+
+namespace sjsu::stm32f10x
+{
+/// An implementation of the sjsu::Pwm peripheral for the stm32f10x family of
+/// microcontrollers.
+class Pwm final : public sjsu::Pwm
+{
+ public:
+  // We use pins spread out across Timers 1 - 5
+  struct PwmPin_t
+  {
+    // Reference to the pin
+    sjsu::Pin & pin;
+
+    // Holds the address to the corresponding Timer peripheral
+    TIM_TypeDef * registers;
+
+    // Holds the channel (1 - 4) which corresponds to the pin
+    uint8_t channel;
+
+    // Peripheral ID for the timer
+    sjsu::SystemController::ResourceID id;
+  };
+
+  struct Channel_t
```

Comment:
```suggestion
  struct Channel  // NOLINT
```

Its an unwritten rule in SJSU-Dev2 to remove the `_t` for namespaces like this inside of a class. Its how the other classes are formatted so this should follow.